requests-for-comments/2022-09-08-notifications.md

```diff
@@ -0,0 +1,81 @@
+# Request for comments: Notifications & @pauldambra 
+
+## Problem statement
+*Who are we building for, what are their needs (include example cases), why is this important?*
+
+Users don't know when to come back to PostHog. Something important can happen and they might miss it. This can be important things in the data, or that another user has edited your insight.
```

Comment:
Hmm, thinking about this a bit more, 

> Users don't know when to come back to PostHog

The fact that they prefer `slack > in-app > email` seems to suggest to me that coming back isn't the full problem here? (if it were, why is there a higher preference to in-app over email notifications?)

But more like 2 different problems based on urgency: (not sure, hypothesising)
1. I want to know when something urgent has happened in `slack / custom webhook` so I can come to PostHog and investigate. Like alerts / when some event is over threshold / someone shared on HN and blog spiked.
2. I want to know what all has changed since I last logged in. These are in-app notifications about things I care about, like others editing my insights, my experiment becoming significant, etc. etc.

Reply:
> I want to know when something urgent has happened in slack / custom webhook so I can come to PostHog and investigate. Like alerts / when some event is over threshold / someone shared on HN and blog spiked.

Notifications can happen in multiple places concurrently. This is where preferences are a huge benefit. You might want some notifications to only go to email, while others you might want to go to slack (because they're more important). You might not want to send notifications for some scenarios. There's a lot of nuance here, so preferences help make the feature flexible enough to account for a broad range of use cases. 

> I want to know what all has changed since I last logged in. These are in-app notifications about things I care about, like others editing my insights, my experiment becoming significant, etc. etc.

This is one place where notifications might not be the best solution. If every change is communicated as a distinct event and has to be cleared from a list, it's going to be really tedious to catch up on changes quickly. This is where something like an activity log can be really useful. Something at a project-level that summarizes important changes since the last time you viewed it. There could be a lot of changes between sessions, so a centralize component might handle that higher volume of changes better. 

Another consideration would be digest notifications - a single notification that covers a broader area of changes. This is great if the user only wants to be notified of changes once day. This case also maps pretty well to generating an email or posting a digest message to slack.